pkg/providers/v1/aws.go

```diff
@@ -4410,8 +4415,22 @@ func (c *Cloud) GetLoadBalancer(ctx context.Context, clusterName string, service
 
 // GetLoadBalancerName is an implementation of LoadBalancer.GetLoadBalancerName
 func (c *Cloud) GetLoadBalancerName(ctx context.Context, clusterName string, service *v1.Service) string {
-	// TODO: replace DefaultLoadBalancerName to generate more meaningful loadbalancer names.
-	return cloudprovider.DefaultLoadBalancerName(service)
+	if val := service.Annotations[ServiceAnnotationLoadBalancerName]; val != "" {
```

Comment:
This doesn't handle existing load balancers provisioned for the service and introduces a potential for leaked LB resources. For example, if you create a service without a custom lb name, controller first provisions a LB with the "a..." name, then later you specify the custom name, controller doesn't clean up the previously created load balancer resource. Previous LB resource get leaked everytime you edit the custom name as well.

Reply:
@kishorj OK, so because the existing behaviour is static -- you cannot change the LB name from the service UID -- that essentially means that we're stuck with non-meaningful LB names, yes? Unless we check for a previous name and delete the LB with the previous name after creating a new LB with the new name. In other words, we would want cloud-provider-aws to be built like any "normal" reconciler ... grab the latest observed state, compare against the desired state, and reconcile the observed to match the desired. The fact that cloud-provider-aws apparently isn't designed in this way makes changes like this one basically dead in the water, no? Sounds like a complete redesign of cloud-provider-aws is needed... :(